dvc/repo/imp_url.py

```diff
@@ -10,6 +14,15 @@ def imp_url(self, url, out=None, fname=None, erepo=None, locked=True):
 
     out = resolve_output(url, out)
 
+    out_dir = os.path.dirname(out or "")
+    # try creating an output directory if it does not already exist
+    if (
+        out_dir
+        and path_isin(os.path.abspath(out_dir), self.root_dir)
+        and not os.path.exists(out_dir)
+    ):
+        makedirs(out_dir, exist_ok=True)
```

Comment:
@skshetry But not creating a dir is expected behavior. Same as when you cp/mv something, it doesn't pre-create things for you. In the issue we've discussed that this is expected behaviour, just need to look into the error message about stage path not existing

Reply:
@efiop, to be more clear, the output should have been:
`file <file> does not exist`, right? There should not be any mention of  `stage working` at all, right?

Reply:
maybe,`Stage.create(wdir=os.curdir, ...)` is what I'm looking for then.

Reply:
@skshetry IIRC, we've discussed this ticket as the one that likely just needs better error messages. `stage working or file path` clearly is broken.